From 0eab630c036dd4ea8d8695317859be6c3e889bbf Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 May 2026 18:35:39 +0800
Subject: [PATCH] dts: arm: ast2705: clean up XGMAC properties

Drop properties from the AST2705 XGMAC node that are not needed by
U-Boot, including the extra compatible, interrupt, FIFO depth, and TSO
settings. Set snps,fb in the AXI setup so the driver can program fixed
burst mode from the device tree.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I6ff03e8782a033540ceb68b382f4221ef29107ee
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/dts/ast2705.dtsi | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/arm/dts/ast2705.dtsi b/arch/arm/dts/ast2705.dtsi
index ee3174d627..2ada158cfc 100644
--- a/arch/arm/dts/ast2705.dtsi
+++ b/arch/arm/dts/ast2705.dtsi
@@ -8,14 +8,9 @@
 
 	soc1: soc@14000000 {
 		xgmac: ethernet@140a0000 {
-			compatible = "aspeed,ast2700-dwxgmac", "intel,socfpga-dwxgmac",
-				     "snps,dwxgmac-2.10", "snps,dwxgmac";
+			compatible = "aspeed,ast2700-dwxgmac", "snps,dwxgmac-2.10", "snps,dwxgmac";
 			reg = <0x0 0x140a0000 0x0 0x8000>;
-			interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "macirq";
 			phy-mode = "10gbase-r";
-			rx-fifo-depth = <4096>;
-			tx-fifo-depth = <4096>;
 			clocks = <&soc1_clk SCU1_CLK_GATE_MAC2CLK>;
 			clock-names = "stmmaceth";
 			resets = <&soc1_rst SCU1_RESET_MAC2>,
@@ -24,7 +19,6 @@
 			snps,axi-config = <&xgmac_axi_setup>;
 			snps,mtl-rx-config = <&xgmac_mtl_rx_setup>;
 			snps,mtl-tx-config = <&xgmac_mtl_tx_setup>;
-			snps,tso;
 			snps,pbl = <8>;
 			phy-handle = <&xpcs0>;
 			pcs-handle = <&xpcs0>;
@@ -34,6 +28,7 @@
 				snps,wr_osr_lmt = <0xf>;
 				snps,rd_osr_lmt = <0xf>;
 				snps,kbbe;
+				snps,fb;
 				snps,blen = <0 0 0 0 0 8 4>;
 			};
 
-- 
2.43.0

